Modules/Loadable/Models/Widgets/qMRMLModelDisplayNodeWidget.cxx

```diff
@@ -633,16 +633,23 @@ void qMRMLModelDisplayNodeWidget::updateWidgetFromMRML()
   double newMin = 0.0;
   double newMax = 0.0;
   int decimals = 0;
-  if (displayRange[0]<displayRange[1])
+  if (displayRange[0] < displayRange[1])
     {
+    // Begin with a precision of 1% of the range
     precision = displayRange[1]/100.0 - displayRange[0]/100.0;
+    // Extend min/max by 20% to give some room
     newMin = (floor(displayRange[0]/precision) - 20 ) * precision;
     newMax = (ceil(displayRange[1]/precision) + 20 ) * precision;
+    // Use closest power of ten value as a step value
```

Comment:
Consider using `ctk::closestPowerOfTen`
See https://github.com/commontk/CTK/blob/master/Libs/Core/ctkUtils.h#L114

Reply:
Adn for the `decimals`, would it work to use `ctk::significantDecimals` ?

Reply:
Nice, that behavior is unwanted though:
https://github.com/commontk/CTK/blob/master/Libs/Core/ctkUtils.h#L109

Reply:
done

Reply:
let me investigate that more, could make a good use of `significantDecimals`

Reply:
I think I have a satisfying solution now